components/benefit_sponsors/app/helpers/benefit_sponsors/application_helper.rb

```diff
@@ -127,6 +127,10 @@ def total_active_census_employees(profile_id)
       employer_profile.census_employees.active.count
     end
 
+    def profile_unread_messages_count(profile)
```

Comment:
Method to differentiate between **model_object** and **form_object** and queries accordingly to get the count of unread messages.